src/pages/Scholarships.jsx

```diff
@@ -33,54 +34,48 @@ function ScholarshipsPage() {
   const location = useLocation();
   const history = useHistory();
 
-  // sets min and max amount for the amount filter
-  const setMinMax = (index, val) => {
+  const params = queryString.parse(location.search, { parseNumbers: true });
+  const { minAmount, maxAmount } = params;
+
+  const setQueryParam = (index, val) => {
     history.push({
       search: queryString.stringify({
-        ...queryString.parse(location.search),
+        ...params,
         [index]: val,
       }),
     });
   };
 
-  const params = queryString.parse(location.search, { parseNumbers: true });
-  const minParam = params.minAmount;
-  const maxParam = params.maxAmount;
-
-  if ((minParam && !Number.isInteger(minParam)) || minParam === 0) {
-    delete params.minAmount;
-    history.replace({ search: queryString.stringify(params) });
-  }
-
-  if ((maxParam && !Number.isInteger(maxParam)) || maxParam === 0) {
-    delete params.maxAmount;
-    history.replace({ search: queryString.stringify(params) });
-  }
+  const clearQueryParam = (index) => {
+    const newParams = { ...params };
+    delete newParams[index];
+    history.replace({ search: queryString.stringify(newParams) });
```

Comment:
```suggestion
    delete params[index];
    history.replace({ search: queryString.stringify(params) });
```

Reply:
That's interesting, before when I tried this I was getting the no-param-reassign eslint warning.